File: src/ProjectR.Scripting/content/scripts/spells/lua/HellsInferno.lua
function GetTargetType()
  return Enemies
end

function GetName()
  return "Hell's Inferno"
end

function GetDescription()
  return "An inferno from hell appears below the enemies.\nDecreases MR."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1.12
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  local damage = (4.5 * aMD * (aFIR/100) - 1.5 * dMR) * (100/dFIR)  
  d:BuffStat(MR, -.25)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/HellsInferno.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class HellsInferno : SpellScriptBase
    {
        public override string Name { get { return "Hell's Inferno"; } }
        public override string Description { get { return "An inferno from hell appears below the enemies.\nDecreases MR."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 112; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4.5 * cMD * (cFIR / 100) - 1.5 * tMR) * (100 / tFIR);
            BuffStat(Stat.MR, -.25);
            DealDamage(damage);
        }
    }
}